cpp/cmake/Modules/ConfigureGoogleBenchmark.cmake

```diff
@@ -0,0 +1,49 @@
+set(GBENCH_ROOT "${CMAKE_BINARY_DIR}/googlebenchmark")
```

Comment:
I know you just copied this file from cudf / rmm, but this file is kinda janky because we wanted to force GBench to be built at configure time instead of at built time, and in hindsight that was probably a mistake. We should probably use an `ExternalProject_Add` command instead and then use the `benchmark` target that is produced from that command for the rest of the cmake.

Reply:
After digging some more, turns out this actually isn't as bad practice as previously thought. Grabbing / building a package during configuration is the only way to be able to use cmake exports from that package, which in this case we'd like to take advantage of.